test/invariant/actors/Proposer.t.sol

```diff
@@ -0,0 +1,117 @@
+        veANGLEDelegation = _veANGLEDelegation;
+    }
+
+    function propose(uint256 value) public useActor(1) {
+        address[] memory targets = new address[](1);
+        bytes[] memory datas = new bytes[](1);
+        uint256[] memory values = new uint256[](1);
+        string memory description = "Test Proposal";
+        targets[0] = address(_angleGovernor);
+        datas[0] = "";
+        values[0] = value;
+
+        uint256 proposalId = _angleGovernor.hashProposal(targets, values, datas, keccak256(bytes(description)));
+        if (_angleGovernor.proposalSnapshot(proposalId) != 0) {
+            vm.expectRevert(
+                abi.encodeWithSelector(
+                    IGovernor.GovernorUnexpectedProposalState.selector,
+                    proposalId,
+                    _angleGovernor.state(proposalId),
+                    bytes32(0)
+                )
+            );
+        }
+        _angleGovernor.propose(targets, values, datas, description);
+
+        // Add to the store
+        proposalStore.addProposal(targets, values, datas, keccak256(bytes(description)));
+    }
+
+    function shortCircuit(uint256 proposalId) public useActor(1) {
+        if (proposalStore.nbProposals() == 0) {
+            return;
+        }
+        Proposal memory proposal = proposalStore.getRandomProposal(proposalId);
+        uint256 proposalHash = _angleGovernor.hashProposal(
+            proposal.target,
+            proposal.value,
+            proposal.data,
+            proposal.description
+        );
+        IGovernor.ProposalState currentState = _angleGovernor.state(proposalHash);
+        if (currentState != IGovernor.ProposalState.Active) {
+            return;
+        }
+
+        uint256 timeElapsed = _angleGovernor.votingDelay() + 1;
+        uint256 blocksElapsed = (_angleGovernor.votingDelay() + 1) / 12;
+        vm.warp(block.timestamp + timeElapsed);
+        vm.roll(block.number + blocksElapsed + 1);
+
+        vm.mockCall(
+            address(veANGLEDelegation),
+            abi.encodeWithSelector(veANGLEDelegation.getPastVotes.selector, _currentActor),
+            abi.encode(_angleGovernor.shortCircuitThreshold(_angleGovernor.proposalSnapshot(proposalHash)) + 1)
+        );
+        _angleGovernor.castVote(proposalHash, 1);
+        _angleGovernor.execute(proposal.target, proposal.value, proposal.data, proposal.description);
+    }
+
+    function execute(uint256 proposalId) public useActor(1) {
+        if (proposalStore.nbProposals() == 0) {
+            return;
+        }
+        Proposal memory proposal = proposalStore.getRandomProposal(proposalId);
+        uint256 proposalHash = _angleGovernor.hashProposal(
+            proposal.target,
+            proposal.value,
+            proposal.data,
+            proposal.description
+        );
+        uint256 proposalSnapshot = _angleGovernor.proposalSnapshot(proposalHash);
+        vm.warp(_angleGovernor.proposalDeadline(proposalHash) + 1);
```

Comment:
I prefer to let it like this and maybe add another function that just execute without using a warp as it will be very very unlikely it will achieve to execute a single proposal with multiple proposals running at the same time